R/GeospatialFunctions.R

```diff
@@ -511,203 +975,644 @@ TADA_ViewATTAINS <- function(.data) {
+              "<br> Assessment Unit Type: ", polygons_mapper$type,
+              "<br> <a href=", polygons_mapper$waterbodyreportlink, " target='_blank'>ATTAINS Link</a>"
+            )
+          ),
+        silent = TRUE
+      )
+
+      # Add ATTAINS lines features (if they exist):
+      try(
+        map <- map %>%
+          leaflet::addPolylines(
+            data = lines_mapper,
+            color = ~ lines_mapper$col,
+            weight = 4, fillOpacity = 1,
+            popup = paste0(
+              "Assessment Unit Name: ", lines_mapper$assessmentunitname,
+              "<br> Assessment Unit ID: ", lines_mapper$assessmentunitidentifier,
+              "<br> Status: ", lines_mapper$overallstatus,
+              "<br> Assessment Unit Type: ", lines_mapper$type,
+              "<br> <a href=", lines_mapper$waterbodyreportlink, " target='_blank'>ATTAINS Link</a>"
+            )
+          ),
+        silent = TRUE
+      )
+
+      # Add ATTAINS point features (if they exist):
+      try(
+        map <- map %>%
+          leaflet::addCircleMarkers(
+            data = points_mapper,
+            lng = ~X, lat = ~Y,
+            color = ~ points_mapper$col, fillColor = ~ points_mapper$col,
+            fillOpacity = 1, stroke = TRUE, weight = 1.5, radius = 5,
+            popup = paste0(
+              "Assessment Unit Name: ", points_mapper$assessmentunitname,
+              "<br> Assessment Unit ID: ", points_mapper$assessmentunitidentifier,
+              "<br> Status: ", points_mapper$overallstatus,
+              "<br> Assessment Unit Type: ", points_mapper$type,
+              "<br> <a href=", points_mapper$waterbodyreportlink, " target='_blank'>ATTAINS Link</a>"
+            )
+          ),
+        silent = TRUE
+      )
+
+      # Add WQP observation features (should always exist):
+      try(
+        map <- map %>%
+          leaflet::addCircleMarkers(
+            data = sumdat,
+            lng = ~LongitudeMeasure, lat = ~LatitudeMeasure,
+            color = "grey", fillColor = "black",
+            fillOpacity = 0.8, stroke = TRUE, weight = 1.5, radius = 6,
+            popup = paste0(
+              "Site ID: ", sumdat$MonitoringLocationIdentifier,
+              "<br> Site Name: ", sumdat$MonitoringLocationName,
+              "<br> Measurement Count: ", sumdat$Sample_Count,
+              "<br> Visit Count: ", sumdat$Visit_Count,
+              "<br> Characteristic Count: ", sumdat$Parameter_Count,
+              "<br> ATTAINS Assessment Unit(s): ", sumdat$ATTAINS_AUs
+            )
+          ),
+        silent = TRUE
+      )
+
+      if (is.null(ATTAINS_lines) & is.null(ATTAINS_points) & is.null(ATTAINS_polygons)) {
+        print("No ATTAINS data associated with this Water Quality Portal data.")
+      }
+
+      # Return leaflet map of TADA WQ and its associated ATTAINS data
+      return(map)
+    }))
+  }
+
 }
+
+# pH data in Larimer County, Colorado for the year 2020.
```

Comment:
@kathryn-willi should this be commented out as well?

comment out TADA_ViewATTAINS lines 1485-1493 and then run devtools::document

Reply:
oh goodness, that was definitely an accident! Will work on pushing these changes to this PR ASAP!

Reply:
Hi @cristinamullin, the previous errors should now be sorted out. Please let me know if you encounter any more and apologies for the mistakes earlier!